py/specter/psf/gausshermite.py

```diff
@@ -311,3 +412,50 @@ def _value(self,x,y,ispec, wavelength):
 
         return img
 
+    def cache_params(self, specrange, wavelengths):
+        #store in a dict
+        legval_dict = dict()
+        legval_dict['x_cache'] = self.legval_cache(self._x, specrange, wavelengths)
+        legval_dict['y_cache'] = self.legval_cache(self._y, specrange, wavelengths)
+        legval_dict['sigx1_cache'] = self.legval_cache(self.coeff['GHSIGX'], specrange, wavelengths)
+        legval_dict['sigy1_cache'] = self.legval_cache(self.coeff['GHSIGY'], specrange, wavelengths)
+        legval_dict['tailxsca_cache'] = self.legval_cache(self.coeff['TAILXSCA'], specrange, wavelengths)
+        legval_dict['tailysca_cache'] = self.legval_cache(self.coeff['TAILYSCA'], specrange, wavelengths)
+        legval_dict['tailamp_cache'] = self.legval_cache(self.coeff['TAILAMP'], specrange, wavelengths)
+        legval_dict['tailcore_cache'] = self.legval_cache(self.coeff['TAILCORE'], specrange, wavelengths)
+        legval_dict['tailinde_cache'] = self.legval_cache(self.coeff['TAILINDE'], specrange, wavelengths)
+        #some extra steps to cache what we need for the core PSF image
+        degx1 = self._polyparams['GHDEGX']
+        degy1 = self._polyparams['GHDEGY']
+        for i in range(degx1+1):
+            for j in range(degy1+1):
+                core_string = 'GH-{}-{}'.format(i,j)
+                legval_dict[core_string]=self.legval_cache(self.coeff[core_string], specrange, wavelengths)
```

Comment:
This change has been implemented as suggested. We now populate self.legval_dict[key] using keys in self.coeff rather than manually specifying them. It should now be more robust.